Google.Api.Gax.Grpc/CallSettings.cs

```diff
@@ -259,7 +260,16 @@ public static CallSettings FromHeader(string name, string value)
             GaxPreconditions.CheckNotNull(name, nameof(name));
             CallSettingsExtensions.CheckHeader(name);
             GaxPreconditions.CheckNotNull(value, nameof(value));
-            return FromHeaderMutation(metadata => metadata.Add(name, value));
+            return name switch
```

Comment:
I'd prefer to put the special-casing in FromGoogleRequestParamsHeader. Yes, this means any other code calling `FromHeader("x-goog-params", ...)` won't get the changed behavior, but I think that's okay.